module3/lessons/week3/CreatingRelatedResources.md

```diff
@@ -0,0 +1,349 @@
+| /artists/:id                          | GET        | Artists#Show      |  
+| /artists/new                          | GET        | Artists#New       |  
+| /artists                              | POST       | Artists#Create    |  
+| /artists/:id/edit                     | GET        | Artists#Edit      |  
+| /artists/:id                          | PUT        | Artists#Update    |  
+| /artists/:id                          | DELETE     | Artists#Destroy   |  
+| /artists/:artistId/paintings          | GET        | Paintings#Index   |  
+| /artists/:artistId/paintings/:id      | GET        | Paintings#Show    |  
+| /artists/:artistId/paintings/new      | GET        | Paintings#New     |  
+| /artists/:artistsId/paintings         | POST       | Paintings#Create  |  
+| /artists/:artistId/paintings/:id/edit | GET        | Paintings#Edit    |  
+| /artists/:artistId/paintings/:id      | PUT        | Paintings#Update  |  
+| /artists/:artistId/paintings/:id      | DELETE     | Artists#Destroy   |  
+
+Note that the routes are _nested_ to reflect the database relationships that exists, but the controllers are not nested.
+
+</section>
+
+## Use TDD to Create a Movie's Review Page
+
+Today, we are going to use TDD to help us drive the development of our new feature!
+
+<section class='call-to-action' markdown='1'>
+
+With a partner, work on creating a test for the User Story below.  Make sure to consider:
+* What objects need to exist in the Arrange section?
+* How will you confirm what should be on the page?
+
+```
+As a user
+When I visit /movies/1/reviews
+Then I see
+    The title of the movie
+    All of that movie's reviews (with a rating and review content)
+```
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### One Solution
+
+```c#
+[Fact]
+public async Task Index_ReturnsViewWithReviews()
+{
+    var context = GetDbContext();
+    var client = _factory.CreateClient();
+
+    Movie spaceballs = new Movie { Genre = "Comedy", Title = "Spaceballs" };
+    Review review1 = new Review { Rating = 5, Content = "Better than Star Wars" };
+    Review review2 = new Review { Rating = 4, Content = "Good. But, when will then be now?" };
+    spaceballs.Reviews.Add(review1);
+    spaceballs.Reviews.Add(review2);
+    Movie youngFrankenstein = new Movie { Genre = "Comedy", Title = "Young Frankenstein" };
+    context.Movies.Add(spaceballs);
+    context.Movies.Add(youngFrankenstein);
+    context.SaveChanges();
+
+    var response = await client.GetAsync($"/Movies/{spaceballs.Id}/Reviews");
+    //Make sure the route exists!
+    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+    var html = await response.Content.ReadAsStringAsync();
+
+    //Make sure the page contains the correct info
+    Assert.Contains(spaceballs.Title, html);
+    Assert.DoesNotContain(youngFrankenstein.Title, html);
+}
+```
+
+</section>
+
+### To make these tests pass, we must:
+1. Create a Review Model
```

Comment:
I'm not sure if the effort is worth it, but I think it would be nice to include the code her for each step, both for students to reference back to and for future instructors. Could maybe even do one big drop down with the steps and the code for them?